### PR TITLE
[Kubevirt] Fixed the test case ResizeVolumeOnScheduleBackup for Kubevirt App

### PR DESCRIPTION
**Pull request by mkoppal-px**
**What this PR does / why we need it**:

- [x] Fixed the test case ResizeVolumeOnScheduleBackup to work with all apps. Earlier it expected the app to have Deployment spec but Kubevirt apps may not have it
- [x] Used function `GetContainerPVCMountMap` to get volume mount points instead of `GetVolumeMounts`
- [x] Added label to kubevirt spec `app: kubevirt-vm-pvc`

**Which issue(s) this PR fixes** (optional)
Closes #PB-4698

**Special notes for your reviewer**:
[Jenkins run with kubevirt app](https://jenkins.pwx.dev.purestorage.com/job/portworx-backup/job/system-tests/job/byoc/job/px-backup-on-demand-system-test-byoc/3173/)
[Jenkins run with postgres](https://jenkins.pwx.dev.purestorage.com/job/portworx-backup/job/system-tests/job/byoc/job/px-backup-on-demand-system-test-byoc/3171)
[Jenkins run with mysql](https://jenkins.pwx.dev.purestorage.com/job/portworx-backup/job/system-tests/job/byoc/job/px-backup-on-demand-system-test-byoc/3172/)

### CODE REVIEW

File: tests/backup/backup_helper.go [hunk 1]
```diff
@@ -979,22 +979,43 @@ func CreateRestoreWithValidation(ctx context.Context, restoreName, backupName st
...
+			break
 		}
 	}
-	number, err = strconv.Atoi(ret)
-	if err != nil {
-		return 0, err
+	if str == "" {
+		log.Infof("Could not find any mount points for the volume mount [%s] in the pod [%s] in namespace [%s] ", volumeMount, podName, namespace)
+		log.Infof("Trying to check if there is a sym link for [%s]", volumeMount)
+		symlinkPath, err := core.Instance().RunCommandInPod([]string{"readlink", "-f", volumeMount}, podName, containerNames[0], namespace)
```
Comment (kphalgun-px) [inline tests/backup/backup_helper.go:998]: Since containerNames is an optional argument how about we check the len of containerNames before we access values in it?

**Review by kphalgun-px — APPROVED** [review]
LGTM

File: tests/backup/backup_helper.go (hunk 1 above)
Comment (mkoppal-px) [reply tests/backup/backup_helper.go]: Fixed it

**Review by dbhatnagar-px — APPROVED** [review]
Lgtm

File: drivers/volume/portworx/schedops/k8s-schedops.go [hunk 2]
```diff
@@ -860,9 +860,9 @@ func (k *k8sSchedOps) GetRemotePXNodes(destKubeConfig string) ([]node.Node, erro
 	return remoteNodeList, nil
 }
 
-// getContainerPVCMountMap is a helper routine to return map of containers in the pod that
+// GetContainerPVCMountMap is a helper routine to return map of containers in the pod that
```
Comment (kshithijiyer-px) [inline drivers/volume/portworx/schedops/k8s-schedops.go:863]: Why is this needed?

Comment (mkoppal-px) [reply drivers/volume/portworx/schedops/k8s-schedops.go:863]: This function was private, I had to make it public to use it in our helper functions.

Comment (kshithijiyer-px) [reply drivers/volume/portworx/schedops/k8s-schedops.go:863]: Got it but one question here is do we need to submit the same fix in schedops as well? if not let's say I vendor in the latest changes in schedops will it revert this fix?

Comment (mkoppal-px) [reply drivers/volume/portworx/schedops/k8s-schedops.go:863]: This is not vendored in code. This is Torpedo code itself.

Comment (kshithijiyer-px) [reply drivers/volume/portworx/schedops/k8s-schedops.go:863]: Okay sounds good.

**Review by kshithijiyer-px — APPROVED** [review]
LGTM
